components/base/sensorcontrolled/spin.go

```diff
@@ -5,97 +5,44 @@ import (
 	"math"
 	"time"
 
-	"github.com/golang/geo/r3"
-	"github.com/pkg/errors"
-	"go.viam.com/utils"
-
-	"go.viam.com/rdk/components/movementsensor"
 	rdkutils "go.viam.com/rdk/utils"
 )
 
 const (
-	increment = 0.01 // angle fraction multiplier to check
-	oneTurn   = 360.0
+	increment   = 0.01 // angle fraction multiplier to check
+	oneTurn     = 360.0
+	slowDownAng = 15. // angle from goal for spin to begin breaking
 )
 
-// Spin commands a base to turn about its center at a angular speed and for a specific angle.
+// Spin commands a base to turn about its center at an angular speed and for a specific angle.
 func (sb *sensorBase) Spin(ctx context.Context, angleDeg, degsPerSec float64, extra map[string]interface{}) error {
-	sb.stopLoop()
-	if math.Abs(angleDeg) >= 360.0 {
-		sb.setPolling(false)
-		sb.logger.CWarn(ctx, "feedback for spin calls over 360 not supported yet, spinning without sensor")
-		return sb.controlledBase.Spin(ctx, angleDeg, degsPerSec, nil)
-	}
-	ctx, done := sb.opMgr.New(ctx)
-	defer done()
-	// check if a sensor context has been started
-	if sb.sensorLoopDone != nil {
-		sb.sensorLoopDone()
+	// if controls are not configured, use the underlying base's spin
+	if len(sb.conf.ControlParameters) == 0 {
+		sb.logger.CWarn(ctx, "control parameters not configured, switching to default spin")
```

Comment:
What is a "default spin"? Some more context around it or a comment explaining what the default spin is would be helpful

Reply:
updated the log to make this clearer, that we intend to use the Spin command of the base that the sensorcontrolled base wraps